frontend/src/metabase/components/AppBanner/AppBanner.unit.spec.tsx

```diff
@@ -0,0 +1,212 @@
+
+import { AppBanner } from "./AppBanner";
+
+interface SetupOpts {
+  shouldRenderPastDueBanner: boolean;
+  shouldRenderUnpaidBanner: boolean;
+  shouldRenderDatabasePromptBanner: boolean;
+}
+
+function setup({
+  shouldRenderPastDueBanner,
+  shouldRenderUnpaidBanner,
+  shouldRenderDatabasePromptBanner,
+}: SetupOpts) {
+  renderWithProviders(<Route path="*" component={AppBanner} />, {
+    initialRoute: "/",
+    storeInitialState: createStateForConfig({
+      shouldRenderPastDueBanner,
+      shouldRenderUnpaidBanner,
+      shouldRenderDatabasePromptBanner,
+    }),
+    withRouter: true,
+  });
+}
+
+function createStateForConfig({
+  shouldRenderPastDueBanner,
+  shouldRenderUnpaidBanner,
+  shouldRenderDatabasePromptBanner,
+}: SetupOpts): State {
+  let isAdmin = false;
+  const tokenStatus = {
+    status: "Token is valid.",
+  };
+  const features = {
+    sso: false,
+  };
+
+  if (shouldRenderPastDueBanner) {
+    tokenStatus.status = "past-due";
+    isAdmin = true;
+  }
+
+  if (shouldRenderUnpaidBanner) {
+    tokenStatus.status = "unpaid";
+    isAdmin = true;
+  }
+
+  if (shouldRenderDatabasePromptBanner) {
+    isAdmin = true;
+    features.sso = true;
+    setupDatabasesEndpoints([createSampleDatabase()]);
+  }
+
+  const state = createMockState({
+    currentUser: createMockUser({ is_superuser: isAdmin }),
+    settings: createMockSettingsState({
+      "token-features": createMockTokenFeatures(features),
+      "token-status": tokenStatus,
+    }),
+  });
+
+  return state;
+}
+
+describe("AppBanner", () => {
+  it.each([
+    {
+      shouldRenderPastDueBanner: false,
+      shouldRenderUnpaidBanner: false,
+      shouldRenderDatabasePromptBanner: false,
+      renderingBanner: "Nothing",
+    },
+    {
+      shouldRenderPastDueBanner: true,
+      shouldRenderUnpaidBanner: false,
+      shouldRenderDatabasePromptBanner: false,
+      renderingBanner: "Past-due banner",
+    },
+    {
+      shouldRenderPastDueBanner: false,
+      shouldRenderUnpaidBanner: true,
+      shouldRenderDatabasePromptBanner: false,
+      renderingBanner: "Unpaid banner",
+    },
+    {
+      shouldRenderPastDueBanner: false,
+      shouldRenderUnpaidBanner: false,
+      shouldRenderDatabasePromptBanner: true,
+      renderingBanner: "Database prompt banner",
+    },
+    {
+      shouldRenderPastDueBanner: true,
+      shouldRenderUnpaidBanner: false,
+      shouldRenderDatabasePromptBanner: true,
+      renderingBanner: "Past-due banner",
+    },
+    {
+      shouldRenderPastDueBanner: false,
+      shouldRenderUnpaidBanner: true,
+      shouldRenderDatabasePromptBanner: true,
+      renderingBanner: "Unpaid banner",
+    },
+  ] as const)(
+    "should render $renderingBanner when should render past-due banner: $shouldRenderPastDueBanner, should render unpaid banner: $shouldRenderUnpaidBanner, should render database prompt banner: $shouldRenderDatabasePromptBanner",
```

Comment:
nit: I might be nit-picking but I don't follow `shouldDoXXX` tests. I think what we actually want to test is different combinations of token features and the `isAdmin` flag, and that you get expected results there. Now the test looks like if `shouldRenderUnpaidBanner` is true, we render the unpaid banner. That's not what we really want to test. Should be fairly straightforward change if you remove the `should` flags and set the assigned properties directly.

Reply:
Other tests are implemented in the way I'd expect, if we can fix this test suite it would be great

Reply:
I was thinking about that too. But since there are tests that cover some combinations of those flags in `DatabasePromptBanner.unit.spec.tsx` and `PaymentBanner.unit.spec.tsx` it feels as if we're repeating a lot of that code here.

To be fair, I was also thinking about removing those 2 test suites and testing every combination here in `AppBanner.unit.spec.tsx` instead. Otherwise, we will just create a superset of tests from those 2 other test suites.

What do you think?